### PR TITLE
Make all paths in Behavior Simulator Twix overlay visible

### PR DESCRIPTION
**Pull request by PasGl**
## Introduced Changes

What it says

## ToDo / Known Issues

none

## Ideas for Next Iterations (Not This PR)

none

## How to Test

Behavior Simulator + Twix

### CODE REVIEW

File: tools/twix/src/panels/map/layers/behavior_simulator.rs [hunk 1]
```diff
@@ -12,23 +12,33 @@ use crate::{
...
+    motion_command: PlayersValueBuffer,
 }
 
 impl Layer for BehaviorSimulator {
     const NAME: &'static str = "Behavior Simulator";
 
     fn new(nao: Arc<Nao>) -> Self {
         let robot_to_field = PlayersValueBuffer::try_new(
-            nao,
+            &nao,
```
Comment (knoellle) [inline tools/twix/src/panels/map/layers/behavior_simulator.rs:23]: Arcs are cheap to clone. Please use them directly instead of a reference.
```suggestion
            nao,clone(),
```

File: tools/twix/src/panels/map/layers/behavior_simulator.rs [hunk 2]
```diff
@@ -39,6 +49,34 @@ impl Layer for BehaviorSimulator {
...
+                                width: 0.025,
+                                color: Color32::from_rgba_unmultiplied(0, 0, 255, 63),
+                            },
+                        ),
+                        PathSegment::Arc(arc, orientation) => painter.arc(
+                            arc,
+                            orientation,
+                            Stroke {
+                                width: 0.025,
+                                color: Color32::from_rgba_unmultiplied(0xAD, 0xD8, 0xE6, 63),
```
Comment (knoellle) [inline tools/twix/src/panels/map/layers/behavior_simulator.rs:73]: Why do you mix hexadecimal and decimal numbers for your colors? Please choose one and stick to it. I'd vote for decimals.

Also, from looking at the code, it is not obvious which color this is, consider defining a named constant and using that here instead.

File: tools/twix/src/panels/map/layers/behavior_simulator.rs [hunk 3]
```diff
@@ -39,6 +49,34 @@ impl Layer for BehaviorSimulator {
...
+
+            if let MotionCommand::Walk { path, .. } = motion_command {
+                for segment in path {
+                    match segment {
+                        PathSegment::LineSegment(line_segment) => painter.line_segment(
+                            robot_to_field * line_segment.0,
+                            robot_to_field * line_segment.1,
+                            Stroke {
+                                width: 0.025,
+                                color: Color32::from_rgba_unmultiplied(0, 0, 255, 63),
```
Comment (knoellle) [inline tools/twix/src/panels/map/layers/behavior_simulator.rs:65]: I think the colors chosen here or rather the low opacity make the lines quite faint.
Personally I would prefer more contrast between these lines and the green field in the background.
But I would still merge this if you are happy with the current state.

File: tools/twix/src/players_value_buffer.rs [hunk 4]
```diff
@@ -10,7 +10,7 @@ use crate::{nao::Nao, value_buffer::ValueBuffer};
 pub struct PlayersValueBuffer(pub Players<ValueBuffer>);
 
 impl PlayersValueBuffer {
-    pub fn try_new(nao: Arc<Nao>, prefix: &str, output: &str) -> Result<Self> {
+    pub fn try_new(nao: &Arc<Nao>, prefix: &str, output: &str) -> Result<Self> {
```
Comment (knoellle) [inline tools/twix/src/players_value_buffer.rs:13]: See above, please revert this change.

**Review by knoellle — CHANGES_REQUESTED** [review]

File: tools/twix/src/panels/map/layers/behavior_simulator.rs (hunk 1 above)
Comment (PasGl) [reply tools/twix/src/panels/map/layers/behavior_simulator.rs]: done

File: tools/twix/src/players_value_buffer.rs (hunk 4 above)
Comment (PasGl) [reply tools/twix/src/players_value_buffer.rs]: done

File: tools/twix/src/panels/map/layers/behavior_simulator.rs (hunk 3 above)
Comment (PasGl) [reply tools/twix/src/panels/map/layers/behavior_simulator.rs]: Opacity has been increased.

File: tools/twix/src/panels/map/layers/behavior_simulator.rs
Comment (knoellle) [inline tools/twix/src/panels/map/layers/behavior_simulator.rs:1]: One further thing to think about: In the map layers I actively decided for rendering the paths before/under the player positions.
This makes it easier to see the robots orientation at the cost of obscuring line paths that are shorter than the robots forward indicator.

I still think the way I described is better, what is your opinion?

Comment (PasGl) [reply tools/twix/src/panels/map/layers/behavior_simulator.rs:1]: I agree.
I think I should also refactor the path-painting into a painter-method to reduce redundant code, now that there is more than one layer painting paths.

Comment (knoellle) [reply tools/twix/src/panels/map/layers/behavior_simulator.rs:1]: Good point, please do.

Comment (PasGl) [reply tools/twix/src/panels/map/layers/behavior_simulator.rs:1]: Both done.

File: tools/twix/src/panels/map/layers/behavior_simulator.rs [hunk 5]
```diff
@@ -38,6 +51,21 @@ impl Layer for BehaviorSimulator {
                 width: 0.02,
                 color: Color32::BLACK,
             };
+
+            let Ok(motion_command): Result<MotionCommand> = self.motion_command.0[player_number].parse_latest() else {
+                continue
+            };
+
+            if let MotionCommand::Walk { path, .. } = motion_command {
```
Comment (knoellle) [inline tools/twix/src/panels/map/layers/behavior_simulator.rs:59]: The let-else-continue here makes it harder to add more stuff later.
What do you think of this?
```suggestion
            if let Ok(MotionCommand::Walk { path, .. }) =
                self.motion_command.0[player_number].parse_latest()
```

Comment (PasGl) [reply tools/twix/src/panels/map/layers/behavior_simulator.rs]: done
